opengrok-web/src/main/webapp/list.jsp

```diff
@@ -65,6 +67,30 @@ org.opengrok.indexer.web.SearchHelper"
     PageConfig cfg = PageConfig.get(request);
     cfg.checkSourceRootExistence();
 
+    String rev = cfg.getRequestedRevision();
+    if (!cfg.isDir() && rev.length() == 0) {
+        /*
+         * Get the latest revision and redirect so that the revision number
+         * appears in the URL.
+         */
+        String location = cfg.getRevisionLocation(cfg.getLatestRevision());
+        if (location != null) {
+            response.sendRedirect(location);
+            return;
+        }
+        if (!cfg.getEnv().isGenerateHtml()) {
+            /*
+             * Economy mode is on and failed to get the last revision
+             * (presumably running with history turned off).  Use dummy
+             * revision string so that xref can be generated from the resource
+             * file directly.
+             */
+            location = cfg.getRevisionLocation(DUMMY_REVISION);
+            response.sendRedirect(location);
+            return;
+        }
+    }
+
     Annotation annotation = cfg.getAnnotation();
```

Comment:
I wonder if annotate works still after the change, especially the combination with revision (get a historical revision and annotate that).

Reply:
Yes, the conditional redirect-to-self was and is dependent on `rev.length == 0`, so there is no redirect if a historical rev is specified.

Reply:
okay, let's try the fix then.